formats/indexed.js

```diff
@@ -0,0 +1,45 @@
+const classic = require('./classic')
+const pify = require('promisify-4loc')
+
+module.exports = function () {
+  return Object.assign(
+    {}, classic, {
+      sbotIsFeed(sbot, author) {
+        const info = sbot.metafeeds.findByIdSync(author)
+        return info && info.feedpurpose === 'index'
+      },
+      appendMsg(sbot, msgVal, cb) {
+        // FIXME: this needs the ability to add 2 messages in a transaction
+        const payload = msgVal.content.indexed.payload
+        delete msgVal.content.indexed.payload
+        sbot.db.add(msgVal, (err, msg) => {
+          if (err) return cb(err)
+          sbot.db.addOOO(payload, (err, indexedMsg) => {
+            if (err) return cb(err)
+            else cb(null, msg)
+          })
+        })
+      },
+      getAtSequence(sbot, pair, cb) {
+        sbot.getAtSequence([pair.id, pair.sequence], (err, msg) => {
+          if (err) return cb(err)
+          const { sequence } = msg.value.content.indexed
+          const authorInfo = sbot.metafeeds.findByIdSync(msg.value.author)
+          if (!authorInfo) return cb(new Error("Unknown author", msg.value.author))
+          const { author } = JSON.parse(authorInfo.metadata.query)
```

Comment:
We could use QL0.parse here, from ssb-subset-ql, which also does validation. might be useful to handle those possible errors